src/components/app/index.js

```diff
@@ -16,24 +14,21 @@ import './app.css';
 class App extends React.Component {
   constructor(props) {
     super(props);
-    const pipelineData = loadData(props.data, this.resetStoreData.bind(this));
-    const initialState = getInitialState(pipelineData, props);
-    this.store = store(initialState);
-    checkFontLoaded(this.store);
+    this.store = configureStore(props);
   }
 
   componentDidUpdate(prevProps) {
     if (prevProps.data.schema_id !== this.props.data.schema_id) {
-      this.resetStoreData(loadData(this.props.data));
+      this.resetStoreData();
     }
   }
 
   /**
    * Dispatch an action to update the store with new pipeline data
-   * @param {Object} formattedData Normalised state data
    */
-  resetStoreData(formattedData) {
-    this.store.dispatch(resetData(formattedData));
+  resetStoreData() {
+    const normalizedData = loadData(this.props.data);
+    this.store.dispatch(resetData(normalizedData));
```

Comment:
I wonder if "normalizedData" could have a better name? I ask because normalising data is usually something like scaling a range of values but in this case loadData is just mapping the data back to a workable structure, what do you think?

Reply:
I think you're talking about [normalization in statistics](https://en.wikipedia.org/wiki/Normalization_(statistics)), as opposed to [database normalization](https://en.wikipedia.org/wiki/Database_normalization)?

Reply:
I named it this because I roughly followed [the guidelines in the Redux docs](https://redux.js.org/recipes/structuring-reducers/normalizing-state-shape/) on state normalization when designing the state when I added a Redux store. Admittedly my state is a lot flatter than the one in the docs.

Reply:
So it doesn't really look like it's normalising the data either way, example from normalzr:

```
[
  {
    id: 1,
    title: 'Some Article',
    author: {
      id: 1,
      name: 'Dan'
    }
  },
  {
    id: 2,
    title: 'Other Article',
    author: {
      id: 1,
      name: 'Dan'
    }
  }
];
```

Can be normalized to:

```
{
  result: [1, 2],
  entities: {
    articles: {
      1: {
        id: 1,
        title: 'Some Article',
        author: 1
      },
      2: {
        id: 2,
        title: 'Other Article',
        author: 1
      }
    },
    users: {
      1: {
        id: 1,
        name: 'Dan'
      }
    }
  }
}
```

But in formatData (which is returning the object) - it's more mapping the loaded data than normalizing it into relational entities e.g.

```
/**
 * Add a new Tag if it doesn't already exist
 * @param {string} name - Default node name
 */
const addTag = state => tag => {
  const { id } = tag;
  state.tags.push(id);
  state.tagName[id] = tag.name;
};
```

Might be worth having a look at the approach here https://github.com/paularmstrong/normalizr/blob/master/docs/quickstart.md and if not the direction you want i'd rename this to avoid confusion about the structure of the state 👍

Reply:
Yeah I appreciate that:
> "Admittedly my state is a lot flatter than the one in the docs."

I'd say that my state is following the approach of the normalized one you've post above, but the only difference is that mine is less deeply nested. e.g. instead of `entities.articles[1].id` I've got something like `articleID[1]`. It's still normalized, it's just not as nested so that updating the state is easier and I don't need to use ImmutableJS because everything is max 1-2 levels deep.

Reply:
From the Redux docs:
> The basic concepts of normalizing data are:
> - Each type of data gets its own "table" in the state.
> - Each "data table" should store the individual items in an object, with the IDs of the items as keys and the items themselves as the values.
> - Any references to individual items should be done by storing the item's ID.
> - Arrays of IDs should be used to indicate ordering.

My approach meets all four of these criteria

Reply:
Ok - must be my wrong interpretation of what's happening, resolving 🙂